internal/configs/bootnode/config.go

```diff
@@ -0,0 +1,252 @@
+var shardConfigs []ConfigType
+var defaultConfig ConfigType
+
+// GetDefaultConfig returns default config.
+func GetDefaultConfig() *ConfigType {
+	return &defaultConfig
+}
+
+// SetDefaultRole ..
+func SetDefaultRole(r Role) {
+	defaultConfig.role = r
+}
+
+func (conf *ConfigType) String() string {
+	return fmt.Sprintf("%v", conf.ShardID)
+}
+
+// SetShardID set the ShardID
+func (conf *ConfigType) SetShardID(s uint32) {
+	conf.ShardID = s
+}
+
+// SetRole set the role
+func (conf *ConfigType) SetRole(r Role) {
+	conf.role = r
+}
+
+// GetShardID returns the shardID.
+func (conf *ConfigType) GetShardID() uint32 {
+	return conf.ShardID
+}
+
+// Role returns the role
+func (conf *ConfigType) Role() Role {
+	return conf.role
+}
+
+// SetNetworkType set the networkType
+func SetNetworkType(networkType NetworkType) {
+	defaultConfig.networkType = networkType
+	for i := range shardConfigs {
+		shardConfigs[i].networkType = networkType
+	}
+}
+
+// GetNetworkType gets the networkType
+func (conf *ConfigType) GetNetworkType() NetworkType {
+	return conf.networkType
+}
+
+// SetVersion set the version of the node binary
+func SetVersion(ver string) {
+	version = ver
+}
+
+// GetVersion return the version of the node binary
+func GetVersion() string {
+	return version
+}
+
+// SetPeerID set the peer ID of the node
+func SetPeerID(pid peer.ID) {
+	peerID = pid
+}
+
+// GetPeerID returns the peer ID of the node
+func GetPeerID() peer.ID {
+	return peerID
+}
+
+// ShardingSchedule returns the sharding schedule for this node config.
+func (conf *ConfigType) ShardingSchedule() shardingconfig.Schedule {
+	return conf.shardingSchedule
+}
+
+// SetShardingSchedule sets the sharding schedule for this node config.
+func (conf *ConfigType) SetShardingSchedule(schedule shardingconfig.Schedule) {
+	conf.shardingSchedule = schedule
+}
+
+// SetShardingSchedule sets the sharding schedule for all config instances.
+func SetShardingSchedule(schedule shardingconfig.Schedule) {
+	defaultConfig.SetShardingSchedule(schedule)
+	for _, config := range shardConfigs {
+		config.SetShardingSchedule(schedule)
+	}
+}
+
+// ShardIDFromKey returns the shard ID statically determined from the input key
+func (conf *ConfigType) ShardIDFromKey(key *bls_core.PublicKey) (uint32, error) {
+	var pubKey bls.SerializedPublicKey
+	if err := pubKey.FromLibBLSPublicKey(key); err != nil {
+		return 0, errors.Wrapf(err,
+			"cannot convert libbls public key %s to internal form",
+			key.SerializeToHexStr())
+	}
+	epoch := conf.networkType.ChainConfig().StakingEpoch
+	numShards := conf.shardingSchedule.InstanceForEpoch(epoch).NumShards()
+	shardID := new(big.Int).Mod(pubKey.Big(), big.NewInt(int64(numShards)))
+	return uint32(shardID.Uint64()), nil
+}
+
+// ValidateConsensusKeysForSameShard checks if all consensus public keys belong to the same shard
+func (conf *ConfigType) ValidateConsensusKeysForSameShard(pubkeys multibls.PublicKeys, sID uint32) error {
```

Comment:
i don't think this is used by the bootnode right ?

Reply:
removed